### PR TITLE
Don't instrument TestKit classpath if not requested

### PR DESCRIPTION
**Pull request by asodja**
Fixes https://github.com/gradle/gradle/issues/28729

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test QF

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java [hunk 1]
```diff
@@ -67,11 +68,29 @@ public DefaultInjectedClasspathPluginResolver(
...
-        this.pluginRegistry = Lazy.locking().of(() -> createPluginRegistry(dependencyResolutionServicesFactory));
-        maybeReportAgentUsageWithTestKitProblem(instrumentationStrategy);
+        this.pluginRegistry = createPluginRegistry(instrumentationStrategy, dependencyResolutionServicesFactory);
+    }
+
+    private Lazy<PluginRegistry> createPluginRegistry(
+        InjectedClasspathInstrumentationStrategy instrumentationStrategy,
+        Factory<DependencyResolutionServices> dependencyResolutionServicesFactory
+    ) {
+        if (instrumentationStrategy.getTransform() == StandardTransform.None) {
```
Comment (asodja) [inline platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java:78]: 💭 Maybe switch here would be nicer

File: platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java [hunk 2]
```diff
@@ -87,13 +106,6 @@ private PluginRegistry createPluginRegistry(Factory<DependencyResolutionServices
         );
     }
 
-    /**
-     * InstrumentationStrategy will report a problem if the agent is used with TestKit, see ConfigurationCacheInjectedClasspathInstrumentationStrategy class.
```
Comment (mlopatkin) [inline platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java:91]: 💅 can we keep this comment somewhere around `getTransform` call?

**Review by mlopatkin — APPROVED** [review]
LGTM with a few minor suggestions

File: platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java [hunk 3]
```diff
@@ -67,11 +68,29 @@ public DefaultInjectedClasspathPluginResolver(
...
     }
 
-    private PluginRegistry createPluginRegistry(Factory<DependencyResolutionServices> dependencyResolutionServicesFactory) {
+    private PluginRegistry createUninstrumentedPluginRegistry() {
+        return new DefaultPluginRegistry(pluginInspector,
+            parentScope.createChild("injected-plugin", null)
+                .local(injectedClasspath)
+                .lock()
+        );
+    }
```
[the remark this answers is not in the document]
Comment (asodja) [reply platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java]: DOne

File: platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java (hunk 2 above)
Comment (asodja) [reply platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/service/internal/DefaultInjectedClasspathPluginResolver.java:91]: Done

**Comment by asodja** [comment]
@bot-gradle test this

**Pull request merged**